.github/workflows/ci.yml

```
@@ -44,7 +44,8 @@ jobs:
         - 9200:9200
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
```

Same here.

```suggestion
    - uses: actions/checkout@v4.1.1
```

Reference: https://github.com/actions/checkout/releases/tag/v4.1.1